DorisProject/src/app/pages/stock/add-stock/add-stock.component.ts

```diff
@@ -13,24 +14,31 @@ import {MAT_DIALOG_DATA, MatDialogRef} from '@angular/material/dialog';
 })
 export class AddStockComponent implements OnInit {
 
-  // tslint:disable-next-line:ban-types
-  state: String = 'NEW';
+
   @Input() formGroupProduct: FormGroup;
   category: any;
   product: any;
-
+  responseModal: any;
+  isUpdate: boolean;
 
   constructor(
     private formBuilder: FormBuilder,
-
     private  router: Router,
     private addstockservice: AddStockService,
-    private dialog: MatDialog,
+    private dialog: MatDialog
   ) {
+    if (this.router.getCurrentNavigation().extras.state.data != null) {
+      this.product = this.router.getCurrentNavigation().extras.state.data;
+      console.log("edit: ",this.product)
+      this.isUpdate = true;
+    } else {
+      this.isUpdate = false;
+    }
   }
 
   ngOnInit(): void {
     this.formGroupProduct = this.formBuilder.group({
+      id:0,
```

Comment:
Codacy has a fix for the issue: [Insert `·`](https://app.codacy.com/gh/SaraArenasV/DorisProjectFrontEnd/pullRequest?prid=9258681)
```suggestion
      id: 0,
```